### DISCUSSION TITLE
Pyinstaller generated executable successfully but execute fail

### DISCUSSION THREAD

**Discussion by LuoSYLuo** [pyinstaller/pyinstaller#6918] [Help]
I am trying to create an executable file from Python(Pyinstaller). When I import oct2py, it failed. Even if I modify the .spec to include oct2py by collecting all of it, it still failed. But I have created/executed the executable file successfully without importing oct2py. it also can enter debug mode under visual studio 2017 IDE and show the Hello World on my GUI.

****Anythins else I miss with imprting oct2py when I apply pyinstaller to generate executable file?** Can anyone help me to solve it?**

Command: Pyinstaller -D --clean myspec.spec PythonApplication.py

Python 3.9.12(conda)

Pyinstaller: 4.8

Platform: Window 10

IDE: Visual Studio 2017

UPX is available

**partition of myspec file**

```
from PyInstaller.utils.hooks import collect_all, collect_submodules, collect_dynamic_libs
datas, binaries, hiddenimports = collect_all('oct2py')
hiddenimports += collect_submodules('oct2py')

a = Analysis(['PythonApplication1.py'],
         pathex=[],
         binaries=binaries,
         datas=[
('C:\ProgramData\Anaconda3\Lib\site-packages\octave_kernel\kernel.json', 'octave_kernel'), 
('C:\ProgramData\Anaconda3\Lib\site-packages\oct2py','oct2py'), 
('C:\ProgramData\Anaconda3\Lib\site-packages\debugpy', 'debugpy')],
```
**piece of source code, I only do the simple code to generate GUI and display "Hellow World"**

```
import sys
import wx
import winxpgui as wgui
import numpy as np
import matplotlib
if getattr(sys, 'frozen', False):
application_path = sys._MEIPASS
os.environ["OCTAVE_KERNEL_JSON"] = os.path.join(application_path, 'octave_kernel/kernel.json')
import zmq.backend.cython
from oct2py import octave
import pprint

if __name__ == '__main__':
x = np.array([[1, 2], [3, 4]], dtype=float)
x = np.int32(1)
y = np.int32(2)
out1, out2 = octave.testcal(3,4,nout=2)
cell = octave.feval('cell',10,10,10)
pprint.pprint([cell.shape])
pprint.pprint([x, x.dtype])
app = wx.App()
frm = HelloWorld(None, title='Main Window')
frm.Show()
app.MainLoop()
```
It possibly imports oct2py fail even if I collect oct2py, octave_kernel in .spec file.
Error message:
```
Traceback (most recent call last):
  File "PythonApplication.py", line 10, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller\loader\pyimod03_importers.py", line 495, in exec_module
  File "oct2py\__init__.py", line 38, in <module>
   octave = Oct2Py()
  File "oct2py\core.py", line 83, in _init_
   self.restart()
  File "oct2py\core.py", line 533, in restart
   self.engine = OctaveEngine(stdin_handler=self._handle_stdin,
  File "octave_kernel\kernel.py", line 176, in _init_
   self.repl = self._create_repl()
  File "octave_kernel\kernel.py", line 402, in _create_repl
   repl = REPLWrapper(cmd, orig_prompt, change_prompt,
   File "metakernel\replwrap.py", line 88, in _init_
   File "metakernel\replwrap.py", line 112, in set_prompt
   File "pexpect\spawnbase.py", line 343, in expect
   File "pexpect\spawnbase.py", line 372, in expect_list
   File "pexpect\expect.py", line 179, in expect_loop
   File "pexpect\expect.py", line 122, in eof
   pexpect.exceptions.EOF:End Of File(EOF).
<pexpect.popen_spawn.PopenSpawn object at 0x0000016195836BE0>
searcher: searcher_re:
0: re.compile('octave.*>')
```

**Comment by rokm** [comment]
From what I can tell, this looks like the output of `octave.exe` program that `oct2py` runs is not what it expects (or rather, is empty). Do you have `OCTAVE` or `OCTAVE_EXECUTABLE` set correctly (i.e., pointing to correct executable) when you are trying to run the frozen application?

**Reply by LuoSYLuo** [reply]
Thanks for your reminder.
I already set the environment variables to add the OCTAVE_EXECUTABLE as the figure.
Anythings else I still miss?
![image](https://user-images.githubusercontent.com/108506563/177231405-0c245d2d-80c1-406d-a98d-dcf3c7a1dad6.png)

**Comment by LuoSYLuo** [comment]
Hi @rokm ,
Thanks for your reminder again. I have solved my issue as below description.
Add the code:
```
if getattr(sys, 'frozen', False):
    application_path = sys._MEIPASS
    os.environ["OCTAVE_KERNEL_JSON"] = os.path.join(application_path, 'octave_kernel/kernel.json')
    os.environ['OCTAVE_EXECUTABLE'] = shutil.which('octave-cli.exe')#add this code
```
After applying Pyinstaller, generate the dict folder. I put my m-file and .mat file into "dict" folder. oct2py.feval will find the file and .mat to calculate.
